quilt/tools/command.py

```diff
@@ -648,6 +652,18 @@ def install(package, hash=None, version=None, tag=None, force=False):
             )
         )
         pkghash = response.json()['hash']
+    elif len(hash) < 64:
+        print('matching short hash against logs...')
+        response = session.get(
+            "{url}/api/log/{owner}/{pkg}/".format(
+                url=QUILT_PKG_URL,
+                owner=owner,
+                pkg=pkg
+            )
+        )
+        for entry in reversed(response.json()['logs']):
+            if entry['hash'][:len(hash)] == hash:
+                pkghash = entry['hash']
```

Comment:
`break` after you found it?

Also, `else` after the `for` loop to handle hash not found

Reply:
DONE - good catch